helm/oauth2-proxy/README.md

```diff
@@ -211,3 +212,27 @@ extraEnv:
   - name: TEST_ENV_VAR_2
     value: '{{ .Values.tplValue }}'
 ```
+
+## htpasswd - Manage credentials for basic authentication
+
+Oauth2-proxy additionally let's you authenticate against a htpasswd file.
+There are two options to provide basic authentication credentials:
+
+1. Generate a htpasswd string with `htpasswd -B` (bcrypt) or `htpasswd -s` (SHA1) and provide it as a list of strings:
+
+   ```yaml
+   htpasswdFile:
+     enabled: true
+     entries:
+       - testuser:{SHA}EWhzdhgoYJWy0z2gyzhRYlN9DSiv
+   ```
+
+2. Provide usernames and passwords as a list of dictionaries and let helm calculate the bcrypt hashes
+
+   ```yaml
+   htpasswdFile:
+     enabled: true
+     entriesPlain:
+       - username: testuser
+         password: passpass
```

Comment:
Do we really want to encourage users to store plaintext passwords? I'm not sure, as a security focused product, we should be recommending this

Reply:
Hmm I don't see a problem here. There are at least two other input values which are also sensitive:
- config.clientSecret
- config.cookieSecret

The rendered manifest still is a Secret and also the helm state is stored in a Secret of type `helm.sh/release.v1`.

Reply:
Maybe it could be configured with an external secret as is already done with `config.existingSecret`.

Reply:
That would be the last option for us, yes. We then would create an umbrella chart and use this feature:
https://github.com/oauth2-proxy/manifests/blob/92dc431007e3423ea1bd8f20c00cdd5871e2158a/helm/oauth2-proxy/values.yaml#L194-L196

Reply:
The helm chart isn't really my area of expertise, but I see the secrets as a bit different from the usernames/passwords. Having usernames and passwords in plaintext makes it incredibly easy to impersonate various users, having the client-secret and cookie-secret at least means theres some significant work to do to be able to get a valid set of credentials.

If it were me, I wouldn't be including any of these secrets in a file anywhere but instead injecting them through some other mechanism

Reply:
Ahh I see what's your point: Impersonation. Yes okay this might be true. But we want to use this htpasswd feature for peripheral systems like Grafana or other monitoring systems, not users.

Reply:
To me, that is still an opportunity for someone to steal data. Putting something behind auth means there's something sensitive you want to protect, even if it is periphery systems. Having passwords in plain text in files is not best practice IMO